lib/skate/detours/detours.ex

```diff
@@ -5,23 +5,121 @@ defmodule Skate.Detours.Detours do
 
   import Ecto.Query, warn: false
   alias Skate.Repo
-
   alias Skate.Detours.Db.Detour
   alias Skate.Settings.User
 
+  @type t :: %__MODULE__{
```

Comment:
Non-blocking, more of a thought than a specific suggestion: This doesn't feel like the right place for this struct to be defined. Specifically I mean top-level in the main `Detours` module. 

I _think_ I'd sooner see this struct defined either within a sub-module here, or in a separate file.

Happy for that to be a follow-up PR though.

Reply:
Ok, so a `detour` sub-module?

Reply:
Leaving as-is.